### DISCUSSION TITLE
CoreDNS in both Tailscale/Headscale and local network?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [coredns/coredns#6506] [Q&A] [answered]
Hello!

I would like to use CoreDNS to handle wildcard domain serving properly - both inside my network at home, and my Headscale-based tailnet. Basically, I would like to achieve something like this:

```
dns://192.168.2.1 {
  // *.birb.it -> resolve to 192.168.2.1
  // nas.birb.it -> resolve to 192.168.2.128
}
dns://100.64.0.2 {
  // *.birb.it -> 100.64.0.2
  // nas.birb.it -> 100.64.0.11
}
```

I saw that CoreDNS mainly uses RFC files for the zones - but I am not sure if I caught that right...

Thanks and kind regards,
Ingwie

**Comment by chrisohaver** [comment, marked as answer]
CoreDNS's _file_ and _auto_ plugins read RFC zone files, and handle wildcard records.

There are a couple of corner case bugs involving wildcard records (#4290 #4256), both I think involving wildcard interactions with empty non-terminals ... fairly obscure, but otherwise should follow the RFC.

**Reply by IngwiePhoenix** [reply]
So all I would have to do is to provide some RFC compliant files and point CoreDNS at them? Okay, should be doable, now that I have a direction. Thanks a lot!
